app/ui/browser/views/page/page.jsx

```diff
@@ -50,49 +50,24 @@ const WEB_VIEW_CHROME_COLLAPSED_STYLE = Style.registerStyle({
 
 class Page extends Component {
   componentDidMount() {
-    const { page, dispatch, ipcRenderer } = this.props;
+    const { webViewController, page, dispatch, ipcRenderer } = this.props;
     const webview = this.refs.webview;
 
     addListenersToWebView(webview, page, dispatch, ipcRenderer);
 
-    if (!page.guestInstanceId && page.location) {
-      webview.setAttribute('src', fixURL(page.location));
-    }
-  }
-
-  /**
-   * After Page receives new properties, iterate over the new commands
-   * in the queue and execute them.
-   *
-   * @TODO This queue grows indefinitely. We should drain it somehow in a
-   * Reduxy way.
-   */
-  componentDidUpdate({ page }) {
-    const currentPage = this.props.page;
-    if (page.commands.size !== currentPage.commands.size) {
-      for (let i = page.commands.size; i < currentPage.commands.size; i++) {
-        this.executeCommand(currentPage.commands.get(i));
+    const filter = fn => (id, ...args) => {
```

Comment:
I considered that, but still wanted it possibly to listen to all events for tests. But that can be done by emitting both `navigate-back` and `navigate-back-${id}` if we wanted, so will do that